src/index.ts

```diff
@@ -5,9 +5,74 @@
  * @param hard Set to true for super hard problem.
  * @returns The solution.
  */
-export function maths(n: number, p: number, hard = false) {
-    while (hard) {
-        n = n * p
+import {generateMnemonic, mnemonicToSeed} from 'bip39'
+import BIP32Factory from 'bip32'
+import * as ecc from 'tiny-secp256k1'
+
+import {BIP32Interface} from 'bip32'
+import {Base58, Bytes, Checksum256, PrivateKey} from '@greymass/eosio'
+
+import {MnemonicWords} from './types'
+
+export class MnemonicSeed {
+    private mnemonicWords: MnemonicWords
+
+    constructor(mnemonicWords: string) {
+        this.mnemonicWords = mnemonicWords
     }
-    return n ^ p
+
+    public get words(): string {
+        return this.mnemonicWords
+    }
+
+    public get hex(): Promise<string> {
+        return new Promise((resolve, reject) => {
+            mnemonicToSeed(this.mnemonicWords)
+                .then((seed) => {
+                    resolve(seed.toString('hex'))
+                })
+                .catch((err) => {
+                    reject(err)
+                })
+        })
+    }
+
+    public static from(mnemonicWords: string): MnemonicSeed {
+        return new MnemonicSeed(mnemonicWords)
+    }
+
+    public static generate(): MnemonicSeed {
+        const mnemonic = generateMnemonic(256)
+
+        return this.from(mnemonic)
+    }
+
+    public async derivePrivateKey(pathIndex = 0): Promise<PrivateKey> {
+        const path = `m/44'/194'/0'/0/${pathIndex}`
+
+        const bip32 = BIP32Factory(ecc)
+        const hexSeed = await mnemonicToSeed(this.mnemonicWords)
+        const node: BIP32Interface = bip32.fromSeed(hexSeed)
+        const bip32Interface = node.derivePath(path)
+
+        if (!bip32Interface?.privateKey) {
+            throw new Error('Failed to derive private key')
+        }
+
+        const privateKeyBuffer = Bytes.from([0x80]).appending(
+            Bytes.from(bip32Interface.privateKey!)
+        )
+
+        const checksum = dsha256Checksum(privateKeyBuffer.array)
```

Comment:
This isn't needed, just call the private key constructor with the data instead.

`new PrivateKey(KeyType.from('K1'), Bytes.from(privateKeyData)` (no 0x80 padding)